docs/appendix/known-limitations.md

```diff
@@ -0,0 +1,255 @@
+and Miden packaging. Once present, we can open up the FFI for general use.
+
+## Core Miden Functionality
+
+### Dynamic Procedure Invocation
+
+- Status: **Unimplemented**
+- Tracking Issue: [#32](https://github.com/0xPolygonMiden/compiler/issues/32)
+- Release Milestone: [Beta 1](https://github.com/0xPolygonMiden/compiler/milestone/4)
+
+This is a dependency of [Function Call Indirection](#function-call-indirection) described above,
+and is the mechanism by which we can perform indirect calls in Miden. In order to implement support
+for indirect calls in the Wasm frontend, we need underlying support for `dynexec`, which is not yet
+implemented.
+
+This feature adds support for lowering indirect calls to `dynexec` or `dyncall` instructions,
+depending on the ABI of the callee. `dyncall` has an additional dependency on support for
+[Cross-Context Procedure Invocation](#cross-context-procedure-invocation).
+
+A known issue with this feature is that `dyn(exec|call)` consumes a word on the operand stack
+for the hash of the callee being invoked, but this word _remains_ on the stack when entering the
+callee, which has the effect of requiring procedures to have a different ABI depending on whether
+they expect to be dynamically-invoked or not.
+
+Our solution to that issue is to generate stubs which are used as the target of `dyn(exec|call)`,
+the body of which drop the callee hash, fix up the operand stack as necessary, and then uses a
+simple `exec` or `call` to invoke the "real" callee. We will emit a single stub for every function
+which has its "address" taken, and use the hash of the stub in place of the actual callee hash.
+
+### Cross-Context Procedure Invocation
+
+- Status: **Unimplemented**
+- Tracking Issue: [#303](https://github.com/0xPolygonMiden/compiler/issues/303)
+- Release Milestone: [Beta 2](https://github.com/0xPolygonMiden/compiler/milestone/5)
+
+This is required in order to support representing Miden accounts and note scripts in Rust, and
+compilation to Miden Assembly.
+
+Currently, you can write code in Rust that is very close to how accounts and note scripts will
+look like in the language, but it is not possible to actually implement either of those in Rust
+today. The reasons for this are covered in depth in the tracking issue linked above, but to
+briefly summarize, the primary issue has to do with the fact that Rust programs are compiled
+for a "shared-everything" environment, i.e. you can pass references to memory from caller to
+callee, write to caller memory from the callee, etc. In Miden however, contexts are "shared-nothing"
+units of isolation, and thus cross-context operations, such as performing a `call` from a note script
+to a method on an account, are not compatible with the usual calling conventions used by Rust and
+LLVM.
+
+The solution to this relies on compiling the Rust code for the `wasm32-wasip2` target, which emits
```

Comment:
Do you mean `wasm32-wasip1` previously known as `wasm32-wasi`?
```suggestion
The solution to this relies on compiling the Rust code for the `wasm32-wasip1` target, which emits
```

Reply:
No, I actually meant `wasm32-wasip2`, which is the Rust target that compiles for the Wasm component model. The `wasm32-wasip1` target is the renamed `wasm32-wasi` target (which is now deprecated), and corresponds to the `preview1` WASI support, while `wasm32-wasip2` is `preview2`, which contains the transition to the Component Model.